src/nupic/algorithms/Connections.hpp

```diff
@@ -322,7 +324,7 @@ namespace nupic
          *
          * @retval Segment data.
          */
-        SegmentData dataForSegment(const Segment& segment) const;
+        SegmentData& dataForSegment(const Segment& segment);
```

Comment:
The compiler is pretty thorough. Inside a const method you can't return a non-const reference to a member.